Extensions/Physics3DBehavior/PhysicsCar3DRuntimeBehavior.ts

```diff
@@ -0,0 +1,1114 @@
+        this.setWheelRadius(newBehaviorData.wheelRadius);
+      }
+      if (oldBehaviorData.wheelWidth !== newBehaviorData.wheelWidth) {
+        this.setWheelWidth(newBehaviorData.wheelWidth);
+      }
+      if (oldBehaviorData.brakeTorqueMax !== newBehaviorData.brakeTorqueMax) {
+        this.setBrakeTorqueMax(newBehaviorData.brakeTorqueMax);
+      }
+      if (
+        oldBehaviorData.handBrakeTorqueMax !==
+        newBehaviorData.handBrakeTorqueMax
+      ) {
+        this.setHandBrakeTorqueMax(newBehaviorData.handBrakeTorqueMax);
+      }
+      if (
+        oldBehaviorData.hasBackWheelDrive !== newBehaviorData.hasBackWheelDrive
+      ) {
+        this.setBackWheelDrive(newBehaviorData.hasBackWheelDrive);
+      }
+      if (
+        oldBehaviorData.hasFrontWheelDrive !==
+        newBehaviorData.hasFrontWheelDrive
+      ) {
+        this.setFrontWheelDrive(newBehaviorData.hasFrontWheelDrive);
+      }
+      if (
+        oldBehaviorData.pitchRollAngleMax !== newBehaviorData.pitchRollAngleMax
+      ) {
+        this.setPitchRollAngleMax(newBehaviorData.pitchRollAngleMax);
+      }
+      return true;
+    }
+
+    override getNetworkSyncData(): PhysicsCar3DNetworkSyncData {
+      // This method is called, so we are synchronizing this object.
+      // Let's clear the inputs between frames as we control it.
+      this._dontClearInputsBetweenFrames = false;
+
+      return {
+        ...super.getNetworkSyncData(),
+        props: {
+          lek: this._wasLeftKeyPressed,
+          rik: this._wasRightKeyPressed,
+          upk: this._wasForwardKeyPressed,
+          dok: this._wasBackwardKeyPressed,
+          hbk: this._wasHandBreakKeyPressed,
+          asf: this._previousAcceleratorStickForce,
+          ssf: this._previousSteeringStickForce,
+          etm: this._engineTorqueMax,
+          esm: this._engineSpeedMax,
+          ei: this._engineInertia,
+        },
+      };
+    }
+
+    override updateFromNetworkSyncData(
+      networkSyncData: PhysicsCar3DNetworkSyncData
+    ) {
+      super.updateFromNetworkSyncData(networkSyncData);
+
+      const behaviorSpecificProps = networkSyncData.props;
+      this._hasPressedForwardKey = behaviorSpecificProps.upk;
+      this._hasPressedBackwardKey = behaviorSpecificProps.dok;
+      this._hasPressedLeftKey = behaviorSpecificProps.lek;
+      this._hasPressedRightKey = behaviorSpecificProps.rik;
+      this._hasPressedHandBreakKey = behaviorSpecificProps.hbk;
+      this._acceleratorStickForce = behaviorSpecificProps.asf;
+      this._steeringStickForce = behaviorSpecificProps.ssf;
+      this._engineTorqueMax = behaviorSpecificProps.etm;
+      this._engineSpeedMax = behaviorSpecificProps.esm;
+      this._engineInertia = behaviorSpecificProps.ei;
+
+      // When the object is synchronized from the network, the inputs must not be cleared.
+      this._dontClearInputsBetweenFrames = true;
+    }
+
+    getPhysicsPosition(result: Jolt.RVec3): Jolt.RVec3 {
```

Comment:
Should that be private and underscored?

Reply:
It can't be private because it's used by an inner class but I'll add the underscore.